addons/web/static/src/views/fields/selection/json_selection_field.js

```diff
@@ -0,0 +1,48 @@
+import { Component } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { selectionField, SelectionField } from "@web/views/fields/selection/selection_field";
+
+export class JsonSelectionField extends Component {
+    static template = SelectionField.template;
+    static props = SelectionField.props;
+    setup() {
+        this.type = "json";
+    }
+    get options() {
+        return this.rawValue?.selection_values || [];
+    }
+    get rawValue() {
+        return this.props.record.data[this.props.name];
+    }
+    get string() {
+        const selected = this.rawValue?.selection_values.find((o) => o[0] === this.value);
+        return selected ? selected[1] : "";
+    }
+    get value() {
+        return this.rawValue?.selection;
```

Comment:
As far as I understand, the form of the json field is something like
```js
{ selection: "a", selection_values: [["a", "Value A"], ["b", "Value B"]] }
```
For regular selection fields, there's a `selection` key in the field definition, which corresponds to your `selection_values` (and obviously, the "value" of the field is your `selection` key). So to make it a bit more generic or natural, what about
```js
{ value: "a", selection: [["a", "Value A"], ["b", "Value B"]] }
```
?

Reply:
It is indeed designed for your usecase, but I'd say that it should be either completely specific (like you did, but as the field is used in base, you have to put it in web, but then maybe in the webclient folder, with a comment saying it is designed for a specific usecase, and maybe hardcode the field name). Or make it as generic as possible, with the more natural API as possible, and then in web like you did. I'd go for the second option.